### ISSUE TITLE
Freeze until repositories are loaded

### ISSUE THREAD

**Issue by Starcommander** [PhoenicisOrg/phoenicis#2234]
### Expected behavior
While updating repositories (in settings) the progress is shown via rotating dots.
The progress does not freeze the whole playonlinux-app.
### Actual behavior
On upstream version the app freezes as long all repositories are loaded.
This needs very long, when new git repo is added (cloning the repo)
### Steps to reproduce
Build the upstream via mvn clean package.
I also created the flatpak package and installed it, but should make no difference.
Run PlayOnLinux, navigate to settings --> repositories --> refresh.
### System information
* software version (git revision): Upstream
* used operating system: Debian 10
* used Java version: Included in flatpak (11)

**Comment by plata** [comment]
I do not get a freeze if I switch to Library/Settings. However, I can reproduce a freeze if I switch to Containers/Engines. Do you have the same behavior?

**Comment by Starcommander** [comment]
No. After adding a git repo and selecting "refresh" the whole window is frozen for a minute.
I can not select anything.
On version alpha2 you see the rotating dots, and there is also no freeze.

**Comment by plata** [comment]
@madoar could this be related to the GUI changes somehow?

**Comment by Starcommander** [comment]
By the way: When adding a new repository via git, the window is too small.
I can only see Cancel/Finish, when resizing the frame.

**Comment by madoar** [comment]
Yes this could be related to recent UI changes. If I remember correctly I changed the way changes to the repositories are propagated. The question is why does the UI freeze? I think that the reason is most likely that the UI takes a while to process the updated repositories, i.e. the changed applications, scripts etc. before they are displayed. I think we (already) download/fetch the new repository in a separate thread.

**Comment by plata** [comment]
> I think that the reason is most likely that the UI takes a while to process the updated repositories, i.e. the changed applications, scripts etc. before they are displayed. I think we (already) download/fetch the new repository in a separate thread.

I totally agree. I also had the impression that loading the repository was already finished when the freeze happens. Could you investigate this?
